### PR TITLE
executor/*: fix bug caused by point_get and output_offsets

### PR DESCRIPTION
**Pull request by AndreMouche**
Hi,
      This PR fixed bug in `point_get`:
       1.  reset upper_bound to `None` 
       2. continue to scan next range when get None. 
      
Meanwhile, this PR also fix bug caused by `output_offsets`.

@andelf @BusyJay @hicqu @hanfei1991 PTAL

### CODE REVIEW

File: src/coprocessor/dag.rs [hunk 1]
```diff
@@ -91,22 +98,20 @@ impl<'s> DAGContext<'s> {
                        mut first: Executor,
                        statistics: &'s mut Statistics)
                        -> Box<DAGExecutor + 's> {
+        let snapshot = SnapshotStore::new(self.snap, self.req.get_start_ts(), self.isolation_level);
```
Comment (BusyJay) [inline src/coprocessor/dag.rs]: Should name it `store` instead.

File: src/coprocessor/executor/scanner.rs [hunk 2]
```diff
@@ -11,62 +11,60 @@
...
+            try!(self.init_with_range(range));
         }
         let seek_key = self.seek_key.take().unwrap();
         if range.get_start() > range.get_end() {
             return Ok(None);
         }
-        let kv = if self.desc {
-            try!(self.reader.reverse_seek(Key::from_raw(&seek_key), self.start_ts))
+
+        let kv = if let Some(scanner) = self.scanner.as_mut() {
```
Comment (BusyJay) [inline src/coprocessor/executor/scanner.rs]: scanner can't be None.

File: src/coprocessor/executor/index_scan.rs [hunk 3]
```diff
@@ -36,10 +37,8 @@ pub struct IndexScanExecutor<'a> {
 impl<'a> IndexScanExecutor<'a> {
     pub fn new(mut meta: IndexScan,
                key_ranges: Vec<KeyRange>,
-               snapshot: &'a Snapshot,
-               statistics: &'a mut Statistics,
-               start_ts: u64,
-               isolation_level: IsolationLevel)
+               snapshot: SnapshotStore<'a>,
```
Comment (BusyJay) [inline src/coprocessor/executor/index_scan.rs]: Name it `store` instead.

**Comment by AndreMouche** [comment]
@BusyJay @andelf @hicqu @hanfei1991 PTAL agin.

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**